src/Core/Silk.NET.BuildTools/Cpp/Clang.cs

```diff
@@ -202,14 +202,35 @@ static void WriteDiagnosticsThrowOnError(TranslationUnit unit)
             var destInfo = task.ClangOpts.ClassMappings[fileName];
             var indexOfOpenSqBracket = destInfo.IndexOf('[');
             var indexOfCloseSqBracket = destInfo.LastIndexOf(']');
-            var projectName = destInfo.Substring
-                (indexOfOpenSqBracket + 1, indexOfCloseSqBracket - indexOfOpenSqBracket - 1);
-            var className = destInfo.Substring(indexOfCloseSqBracket + 1);
+            var projectNameSplit = destInfo.Substring
+                (indexOfOpenSqBracket + 1, indexOfCloseSqBracket - indexOfOpenSqBracket - 1).Split(':');
+
+            if (projectNameSplit.Length == 0) 
+            {
+                throw new ArgumentException("Missing project name!");
+            }
+
+            if (projectNameSplit.Length > 2)
+            {
+                throw new ArgumentException("Too many splits in the project name!");
+            }
+
+            var projectName = projectNameSplit[0];
+
+            if(projectName != "Core" && coreProfile == null) 
+            {
+                throw new InvalidOperationException("The core profile/source file must come first!");
+            }
```

Comment:
As discussed, Clang and ConvertConstruct must not make assumptions about any other generation operations that happen outside of it. (continued in next comment)